### PR TITLE
Update exact_substrings.py

### PR DESCRIPTION
**Pull request by jordane95**
Fix indent bug

### CODE REVIEW

**Comment by jordane95** [comment]
Hi @guipenedo 

I find this line throws error when the data is empty, i is not defined.

After second thought, this isn't a good fix since it is intended to check it only once at the end of the iteration?

**Comment by guipenedo** [comment]
Thanks for the PR! Indeed, it should be outside the for loop. Maybe just add a `i = 0` above the for loop?

**Comment by jordane95** [comment]
Exactly! But i should be -1 in this case to pass the assertion of doc_lens

**Comment by guipenedo** [comment]
You're right yeah. Could you then please revert your original change and remove the indent for the 2 asserts?

**Comment by jordane95** [comment]
Done! Actually the previous one was already outside the `for` loop but inside the `with` context, should make on difference

File: src/datatrove/pipeline/dedup/exact_substrings.py [hunk 1]
```diff
@@ -313,9 +314,11 @@ def run(self, data: DocumentsPipeline = None, rank: int = 0, world_size: int = 1
         self.rank = rank
         # loads the sequence file from stage 1, the size file from stage 1 and the bytearange file.
         sequence_file, size_file = self.get_all_files(rank=self.rank, world_size=world_size)
+        if not self.dup_ranges:
+            yield None
```
Comment (guipenedo) [inline src/datatrove/pipeline/dedup/exact_substrings.py:318]: maybe just `return` instead?

**Comment by guipenedo** [comment]
I admit it has been a while since I last ran this block, thanks for the changes. Is it currently running well with the new changes? I think the tests for the last block are lacking a bit

File: src/datatrove/pipeline/dedup/exact_substrings.py (hunk 1 above)
Comment (jordane95) [reply src/datatrove/pipeline/dedup/exact_substrings.py:318]: Yeah, `return` works.

**Comment by jordane95** [comment]
Not yet. I will look into it and make some changes soon

**Comment by jordane95** [comment]
Actually this issue is also fixed by early existing.

> does this also fix #76 ?

**Comment by guipenedo** [comment]
Perfect! Thanks a lot for all the work, merged

**Pull request merged**